torch/nn/modules/module.py

```diff
@@ -913,6 +914,10 @@ def parameters(self, recurse=True):
             <class 'torch.Tensor'> (20L, 1L, 5L, 5L)
 
         """
+        # parameters replicated models are not populated to _parameters
+        if self.is_replicated:
```

Comment:
This looks OK to me, but I am not sure which one is better when compared to the following option:

* In `replicate.py`, set `replica.parameters` to a function that raise this error. 

cc @ngimel 
cc @suo @wanchaol if we modify `replica.parameters` when replicating, will that break JIT?

Reply:
@mrshenli so I've been thinking about your comment and can't seem to answer the question on my own. When you say we should set "replica.parameters" to a function, what do you mean? Because as far as I am able to understand, the replica models aren't their own object (i.e. ReplicaModel), they're just models.